src/Core/Form/IdentifiableObject/CommandBuilder/Accessor/CommandAccessor.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor;
+
+use Symfony\Component\PropertyAccess\Exception\NoSuchIndexException;
+use Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * This class is inspired from the PropertyAccessor component from Symfony, but it's centered around
+ * prefilling command objects based on a config and the data available in the flattened form data array.
+ */
+class CommandAccessor
+{
+    /**
+     * @var CommandAccessorConfig
+     */
+    private $config;
+
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(CommandAccessorConfig $config)
+    {
+        $this->config = $config;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessorBuilder()
+            ->enableExceptionOnInvalidIndex()
+            ->enableExceptionOnInvalidPropertyPath()
+            ->disableMagicCall()
+            ->getPropertyAccessor()
+        ;
+    }
+
+    /**
+     * If only single store command is provided then only this one will be prefilled, however if a command for all
+     * stores is also provided then this component will check in the data (for each field) if the modification targets
+     * all the stores, or not, and prefill the appropriate command accordingly.
+     *
+     * @param array $data
+     * @param mixed $singleStoreCommand
+     * @param mixed|null $allStoresCommand
+     *
+     * @return array Returns modified commands (if no field was detected an empty array is returned)
+     */
+    public function buildCommands(
+        array $data,
+        $singleStoreCommand,
```

Comment:
Isn't an array or null?
Can by ?array no?
I said that because of the in_array function

Reply:
No these objects are CQRS commands, like `UpdateProductPricesCommand` for example The `in_array` in this function is used with the command as the needle (`mixed`), not the haystack array

Reply:
Funny that we don't have a default CQRS command interface :thinking:

Reply:
> Funny that we don't have a default CQRS command interface 🤔

Because it would have nothing in it, it would be an empty interface. A Command is... almost a DTO. Almost a StdClass.

Maybe however we could cast `$singleStoreCommand` into a CommandCollection object if @PierreRambaud does not want arrays.

Reply:
But again `$singleStoreCommand` and `$allStoresCommand` are NOT arrays, they are objects. But the returns of this method is an array of commands (even if it can contain one, two or no commands sometimes).

And I need two internal arrays, the second one is used to make sure the commands are always returned in the same order.